File: Golang/506. Relative Ranks.go
func findRelativeRanks(nums []int) []string { // 132ms
    length := len(nums)
    ordered := make([]int, length)
    copy(ordered, nums)
    sort.Ints(ordered)
    
    ranksDict := make(map[int]string, length)
    for i := 1; i <= length; i++ {
        ranksDict[ordered[i-1]] = strconv.Itoa(length - i + 1)
    }
    
    ans := make([]string, 0, length)
    for _, score := range nums {
        switch ranksDict[score] {
        case "1": ans = append(ans, "Gold Medal")
        case "2": ans = append(ans, "Silver Medal")
        case "3": ans = append(ans, "Bronze Medal")
        default: ans = append(ans, ranksDict[score])
        }
    }
    return ans
}

File: Python3/506. Relative Ranks.py
class Solution:
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        ordered = sorted(nums, reverse=True) # 76ms
        ranksDict = dict(zip(ordered, range(1, len(nums)+1)))
        ranks = []
        for num in nums:
            if ranksDict[num] == 1:
                ranks.append("Gold Medal")
            elif ranksDict[num] == 2:
                ranks.append("Silver Medal")
            elif ranksDict[num] == 3:
                ranks.append("Bronze Medal")
            else:
                ranks.append(str(ranksDict[num]))
        return ranks
